gnat2goto/driver/tree_walk.adb

```diff
@@ -695,28 +692,24 @@ package body Tree_Walk is
 
       declare
          LHS : constant Irep := Do_Expression (Name (N));
-         RHS : constant Irep := Do_Expression (Expression (N));
-         R : constant Irep := New_Irep (I_Code_Assign);
+         function RHS return Irep;
+         function RHS return Irep is
+            N_RHS : constant Node_Id := Expression (N);
+            Bare_RHS : constant Irep := Do_Expression (N_RHS);
+         begin
+            return
+              (if Do_Range_Check (N_RHS)
+               then Make_Range_Assert_Expr
+                 (N => N,
+                  Value => Bare_RHS,
+                  Bounds_Type => Get_Type (LHS))
+               else Bare_RHS);
+         end RHS;
       begin
-         Set_Source_Location (R, Sloc (N));
-         Set_Lhs (R, LHS);
-         if Do_Range_Check (Expression (N)) then
-            declare
-               Range_Expr : constant Irep :=
-                 Make_Range_Assert_Expr (
-                          N          => N,
-                          Value      => RHS,
-                          Bounds_Type => Get_Type (LHS));
-            begin
-               Set_Rhs (R,
-                        Typecast_If_Necessary (Range_Expr, Get_Type (LHS),
-                          Global_Symbol_Table));
-            end;
-         else
-            Set_Rhs (R, Typecast_If_Necessary (
-                     RHS, Get_Type (LHS), Global_Symbol_Table));
-         end if;
-         return R;
+         return Make_Code_Assign
+           (Lhs => LHS,
+            Rhs => RHS,
```

Comment:
I'm pretty sure the typecasts were there for a reason, but it's nice it works without them.

Reply:
I didn't intentionally remove them. Ohps. Do you think I should add them back? If the presence or absence of these type casts can make a difference we should have a test to show it, and we apparently don't right now, which is a bit of a shame